contracts/GNSlib.sol

```diff
@@ -0,0 +1,101 @@
+pragma solidity ^0.4.24;
+
+contract GNSlib {
+
+    constructor() public {
+
+    }
+
+    function stringToHexHelper(string what) pure public returns (bytes) {
+        return bytes(what);
+    }
+
+    /** @notice Verifies name validness. Name should not contain dots (".")
+    */
+    function isValidName(string _name) pure public returns (bool) {
+        bytes memory nameInByteArray = bytes(_name);
+        for (uint128 i = 0; i < nameInByteArray.length; i++)
+            if (nameInByteArray[i] == '.')
+                return false;
+        return true;
+    }
+
+    /** @notice Verifies validness of part of record
+    */
+    function isValidString(bytes _str, uint32 _offset, uint32 _length) pure public returns (bool) {
+        if (_str.length < _offset + _length)
+            return false;
+        for (uint128 i = _offset; i < _length; i++)
+            if (_str[i] == 0)
+                return false;
+        return true;
+    }
+
+    /** @notice Verifies is record name exists
+    */
+    function bytesToUint32LE(bytes _what, uint32 _offset) pure public returns (uint32) {
+        require(_what.length >= _offset + 4);
+        return uint32(_what[_offset + 3])
+        | (uint32(_what[_offset + 2]) << 8)
+        | (uint32(_what[_offset + 1]) << 16)
+        | (uint32(_what[_offset + 0]) << 24);
+    }
+
+    /**
+     * If type of protocol not in range for custom or not unknown type, the function returns false
+     */
+    function isValidRecord(bytes _rawRecord) pure public returns (bool) {
```

Comment:
The same thing. No validity checks for now.

Reply:
Here and further...

Reply:
They're not used at this moment